File: qt/gen_qtestsupport_gui.cpp
// +build ignore

#include <QPoint>
#define WORKAROUND_INNER_CLASS_DEFINITION_QTest__QTouchEventSequence
#include <QWindow>
#include <qtestsupport_gui.h>
#include "gen_qtestsupport_gui.h"

QTouchEventSequence* QTest__QTouchEventSequence_Press(QTest__QTouchEventSequence* self, int touchId, QPoint* pt) {
	return &self->press(static_cast<int>(touchId), *pt);
}

QTouchEventSequence* QTest__QTouchEventSequence_Move(QTest__QTouchEventSequence* self, int touchId, QPoint* pt) {
	return &self->move(static_cast<int>(touchId), *pt);
}

QTouchEventSequence* QTest__QTouchEventSequence_Release(QTest__QTouchEventSequence* self, int touchId, QPoint* pt) {
	return &self->release(static_cast<int>(touchId), *pt);
}

QTouchEventSequence* QTest__QTouchEventSequence_Stationary(QTest__QTouchEventSequence* self, int touchId) {
	return &self->stationary(static_cast<int>(touchId));
}

bool QTest__QTouchEventSequence_Commit(QTest__QTouchEventSequence* self, bool processEvents) {
	return self->commit(processEvents);
}

QTouchEventSequence* QTest__QTouchEventSequence_Press3(QTest__QTouchEventSequence* self, int touchId, QPoint* pt, QWindow* window) {
	return &self->press(static_cast<int>(touchId), *pt, window);
}

QTouchEventSequence* QTest__QTouchEventSequence_Move3(QTest__QTouchEventSequence* self, int touchId, QPoint* pt, QWindow* window) {
	return &self->move(static_cast<int>(touchId), *pt, window);
}

QTouchEventSequence* QTest__QTouchEventSequence_Release3(QTest__QTouchEventSequence* self, int touchId, QPoint* pt, QWindow* window) {
	return &self->release(static_cast<int>(touchId), *pt, window);
}

void QTest__QTouchEventSequence_Delete(QTest__QTouchEventSequence* self, bool isSubclass) {
	if (isSubclass) {
		delete dynamic_cast<QTest::QTouchEventSequence*>( self );
	} else {
		delete self;
	}
}

File: qt/gen_qtestsupport_gui.go
package qt

import (
	"unsafe"
)

type QTest__QTouchEventSequence struct {
	h          uintptr
	isSubclass bool
}

func (this *QTest__QTouchEventSequence) Press(touchId int, pt *QPoint) *QTouchEventSequence {
	xxxxxxxxx
}

func (this *QTest__QTouchEventSequence) Move(touchId int, pt *QPoint) *QTouchEventSequence {
	xxxxxxxxx
}

func (this *QTest__QTouchEventSequence) Release(touchId int, pt *QPoint) *QTouchEventSequence {
	xxxxxxxxx
}

func (this *QTest__QTouchEventSequence) Stationary(touchId int) *QTouchEventSequence {
	xxxxxxxxx
}

func (this *QTest__QTouchEventSequence) Commit(processEvents bool) bool {
	return (bool)(QTest__QTouchEventSequence_Commit(this.h, (bool)(processEvents)))
}

func (this *QTest__QTouchEventSequence) Press3(touchId int, pt *QPoint, window *QWindow) *QTouchEventSequence {
	xxxxxxxxx
}

func (this *QTest__QTouchEventSequence) Move3(touchId int, pt *QPoint, window *QWindow) *QTouchEventSequence {
	xxxxxxxxx
}

func (this *QTest__QTouchEventSequence) Release3(touchId int, pt *QPoint, window *QWindow) *QTouchEventSequence {
	xxxxxxxxx
}
